### DISCUSSION TITLE
foot switch with 5 banks and increment decrement selector

### DISCUSSION THREAD

**Discussion by benwadub** [tttapa/Control-Surface#423] [Q&A]
hi, I can t find why I can t switch between banks in my project,
i have 3 footswitch that send cc 1, 2 and 3
2 switch to use as bank increment and decrement selector,
all theses 5 switch are properly wired because I can send cc with each 5 but I can t find why I can t use 2 of them to change my selected bank, it s the first time I try to have more than 2 banks, can you point me where I made a mistake please?
```cpp
/** 
 * This is an example that demonstrates the use of MIDI Control Change Buttons.
 *
 * @boards  AVR, AVR USB, Nano Every, Due, Nano 33, Teensy 3.x, ESP32
 * 
 * Connections
 * -----------
 * 
 * - 5: momentary push button (to ground)
 * 
 * The internal pull-up resistor for the button will be enabled automatically.
 * 
 * Behavior
 * --------
 * 
 * - When the button on pin 5 is pressed, a MIDI Control Change message with a 
 *   value of 0x7F (127) is sent for General Purpose Controller #1.
 * - When the button on pin 5 is released, a MIDI Control Change message with a 
 *   value of 0x00 (0) is sent for General Purpose Controller #1
 * 
 * Mapping
 * -------
 * 
 * Select the Arduino as a custom MIDI controller in your DAW, and use the 
 * MIDI learn option to assign the button to a function.
 * 
 * Written by PieterP, 2019-09-29  
 * https://github.com/tttapa/Control-Surface
 */

#include <Control_Surface.h> // Include the Control Surface library

// Instantiate a MIDI over USB interface.
USBMIDI_Interface midi;

Bank<5> bank(1);

IncrementDecrementSelector<5> selector = {
    bank,       // Bank to manage
    {A5, A4},     // push button pins (increment, decrement)

};
// Instantiate a CCButton object
Bankable::CCButton buttons[] = {
  {{bank,BankType::CHANGE_ADDRESS}, A8, {1, CHANNEL_12}},//numéro de bank correspondant/indique que le changement de bank change le canal midi (+8 car on a 8 adresses par bank)/pin sur laquelle le controleur est branché/numéro de cc/numéro de canal midi
  {{bank,BankType::CHANGE_ADDRESS}, A7, {2, CHANNEL_12}},
  {{bank,BankType::CHANGE_ADDRESS}, A6, {3, CHANNEL_12}},


};

void setup() {
  Control_Surface.begin(); // Initialize Control Surface
}

void loop() {
  Control_Surface.loop(); // Update the Control Surface
}```

**Comment by ziplob123** [comment]
I may be wrong but-
Bank<5> bank(1);
This feels like you want 5 banks but then forcing it to use 1
Maybe try
Bank<5> bank;
or
Bank<5> bank(5);

Sorry if i'm wrong but in my bankable Footpedal i used code has follows-

#include <Control_Surface.h>
HairlessMIDI_Interface stream;
CCButton But1 = { 8, 62};
CCButton But2 = { 4, 63};
CCButton But3 = { 5, 64};
CCButton But5 = { 6, 65};
CCButton But6 = { 2, 66};
CCButton But8 = { 3, 67};
CCButton But9 = { 10, 68};
CCButton But10 = { 11, 69};

Bank<4> bank;
IncrementDecrementSelector<4> selector = {bank, {7, 9}};

using CCSmartPot = Bankable::CCSmartPotentiometer<4>;

CCSmartPot potentiometer = {
{bank, BankType::CHANGE_CHANNEL}, // Bank configuration
A0, // Analog pin connected to potentiometer
{MIDI_CC::Channel_Volume, CHANNEL_1}, // Channel volume of channel 1
};

void setup() {
Control_Surface.begin();
}
void loop()
{
Control_Surface.loop();
}

**Comment by tttapa** [comment]
What kind of switches are you using? The IncrementDecrementSelector class expects momentary push buttons wired between ground and the input pin. Latching foot switches will not work correctly.

**Reply by benwadub** [reply]
hi, I resolved my problem by using the other pole of my footswitch them and in the middle and two others pins, I just swept them and all is fine! sorry for the noob question :-)
